### PR TITLE
Remove react-leaflet

### LINKED ISSUE #1105: Current position is hidden by the buffered positions in Map panel
When using a bag file, the buffering places blue markers on the map panel. When I press play it is not possible to know where the current position is since the map is already populated by blue markers. I recommend to place a marker of different color or size for the current position and small blue markers for the buffered positions.

foxglove

Overall displaying markers for the buffered portion of the bag file is very focusing. If I click on a few different locations in the bag file, the buffering process creates unlinked graphics in plot and map panels.

foxglove2

### LINKED ISSUE #1182: Satellite view for Map panel
Requested in https://github.com/foxglove/studio/discussions/401#discussioncomment-711512

### PR DESCRIPTION
**Pull request by defunctzombie**
react-leaflet uses a non-standard OSS license. Since we use only a small set of leaflet
features we move over to using leaflet directly.

Fixes #1182
Fixes #1105

### CODE REVIEW

File: packages/studio-base/src/panels/Map/MapPanel.tsx [hunk 1]
```diff
@@ -49,14 +53,44 @@ function MapPanel(props: MapPanelProps): JSX.Element {
...
+      return;
+    }
+
+    const tileLayer = new TileLayer("https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png", {
+      attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+      maxNativeZoom: 18,
+      maxZoom: 24,
+    });
+
+    const satelite = new TileLayer(
```
Comment (jtbandes) [inline packages/studio-base/src/panels/Map/MapPanel.tsx:68]: typo: satellite

**Review by jtbandes — COMMENTED** [review]
If this PR added features (satellite view) it should be called out more explicitly in the PR description - as is it's not clear that this PR contains anything changelog-worthy

File: packages/studio-base/src/panels/Map/MapPanel.tsx (hunk 1 above)
Comment (defunctzombie) [reply packages/studio-base/src/panels/Map/MapPanel.tsx:68]: Will make a PR for seek preview (hover value) and fix that there. Spelling is hard for me.

Comment (jtbandes) [reply packages/studio-base/src/panels/Map/MapPanel.tsx:68]: There's a VS Code Extension For That™

**Pull request merged**